src/app/calculator/furnaces/o2-enrichment/o2-enrichment.component.ts

```diff
@@ -90,106 +76,54 @@ export class O2EnrichmentComponent implements OnInit {
     }, 100);
   }
 
-  btnResetData() {
-    if (this.inAssessment) {
-      this.calculator = this.originalCalculator;
-    } else {
-      this.o2Form = this.o2EnrichmentService.initFormFromObj(this.settings, this.o2EnrichmentService.getResetData());
-    }
-    this.lines = [];
-    if (this.o2EnrichmentService.lines) {
-      this.o2EnrichmentService.lines = [];
-    }
-    this.calculate();
-    this.toggleResetData = !this.toggleResetData;
-  }
-
-  generateExample() {
-    this.o2Enrichment = this.o2EnrichmentService.generateExample(this.settings);
-    this.o2Form = this.o2EnrichmentService.initFormFromObj(this.settings, this.o2Enrichment);
-    this.calculate();
-  }
-
-  btnGenerateExample() {
-    this.generateExample();
-    this.calculate();
+  ngOnDestroy() {
+    this.enrichmentInputsSub.unsubscribe();
   }
 
-
-  resizeTabs() {
-    if (this.leftPanelHeader.nativeElement.clientHeight) {
-      this.headerHeight = this.leftPanelHeader.nativeElement.clientHeight;
-    }
+  initSubscriptions() {
+    this.enrichmentInputsSub = this.o2EnrichmentService.enrichmentInputs.subscribe(value => {
+      this.enrichmentInputs = value;
+      this.calculate();
+    })
   }
 
   calculate() {
-    this.o2Enrichment = this.o2EnrichmentService.getObjFromForm(this.o2Form);
```

Comment:
Don't really need the calculate function that just the .next on the service. And commented out code can be deleted